api/src/Responder/Responder/ApiResponder.php

```diff
@@ -137,16 +143,65 @@ protected function collectionWithStatusCode($items, $code = 200)
     }
 
     /**
-     * TODO: Implement paginator() method.
-     * Bind a paginator to a transformer and start building a response.
+     * Build paginated response.
      *
-     * @param Paginator $paginator
+     * @param Collection|array $items
+     * @param null|int $offset
+     * @param null|int $totalCount
      * @param array $parameters
      *
      * @return Response
      */
-    public function paginator(Paginator $paginator, array $parameters = [])
+    public function paginatedCollection($items, $offset = null, $totalCount = null, array $parameters = [])
+    {
+        $itemCount = count($items);
+        $this->validateRange($itemCount);
+
+        $response = $this->getResponse();
+        $response->headers->set('Accept-Ranges', 'entities');
+        $response->headers->set('Content-Type', 'application/json');
+        $response->setStatusCode(206);
+
+        $rangeHeader = $this->prepareRangeHeader($itemCount, $offset, $totalCount);
+        $response->headers->set('Content-Range', $rangeHeader);
+
+        $response->setContent($this->encode($this->getTransformer()->transformCollection($items)));
+
+
+
+        return $response;
+    }
+
+    /**
+     * @param $itemCount
+     * @return bool
+     */
+    protected function validateRange($itemCount)
     {
+        if ($this->request->headers && $this->request->headers->has('Range')) {
+            if ($itemCount <= 0) {
+                throw new HttpException(416, 'Requested Range Not Satisfiable');
+            }
+        } else {
+            throw new HttpException(400, 'Bad Request');
+        }
+        return true;
+    }
```

Comment:
I think, that `Requested Range Not Satisfiable`, when the query is totally out of range.
If at least one record is returned, then it's ok.

Ex. 
Request
Range: entities 199-219
Response: entities 199-204/* 

This will give a valid response on one hand and the number of entities on the other. It's good in case we don't want to return total number of records